presto-main/src/main/java/com/facebook/presto/metadata/FunctionManager.java

```diff
@@ -61,6 +62,11 @@ public void addFunctions(List<? extends SqlFunction> functions)
         return globalFunctionNamespace.listFunctions();
     }
 
+    public FunctionHandle resolveFunction(Session session, QualifiedName name, List<TypeSignatureProvider> parameterTypes)
```

Comment:
It's introduced in @dain's original PR: https://github.com/prestosql/presto/pull/196/commits/7915b629aa2c36a88c6b21a705b6650617e3525b. A technical reason to add it now (even it's not used) is otherwise the two `resolveFunction` would have exactly the same signature. Though I don't think we will give SQL functions access to `session`, I think `session` will just be used to resolve which function to use, so it will only be used within `FunctionManager`/`FunctionNamespace`.

Reply:
SqlPath maybe more favorable here. Since we are not using Session(or SqlPath) at the moment anyway, this may as well be just an interface?